Snowiiii/Pumpkin
# Added basic chest opening and closing

## Description
Added chest opening and closing animation and only trigger sound when the first or last player does the correct action.


## Testing
Two clients:

- Place one chest, open and close, see animation in both and notice sound
- Open in second client, close in first, stays open and no sound
- Close in both, it closes and plays a sound

This does not add the constant ticks and also sends all chest packets to all players instead of just when the respective chunks are loaded. Both of these issues would be separate PRs.


## Checklist
Things need to be done before this Pull Request can be merged.

- [x] All unit tests pass: `cargo test`
- [x] I added new unit tests, so other people don't accidentally break my code by changing other parts of the codebase. [How?](https://doc.rust-lang.org/book/ch11-01-writing-tests.html)
